py7zr/py7zr.py

```diff
@@ -644,15 +644,16 @@ def test(self) -> bool:
         """Test archive using CRC digests."""
         return self._test_digests()
 
-    def extractall(self, path: Optional[Any] = None) -> None:
+    def extractall(self, path: Optional[Any] = None, return_dict: bool = False) -> Optional[Dict[str, IO[Any]]]:
         """Extract all members from the archive to the current working
            directory and set owner, modification time and permissions on
            directories afterwards. `path' specifies a different directory
            to extract to.
         """
-        return self.extract(path)
+        return self.extract(path, return_dict=return_dict)
 
-    def extract(self, path: Optional[Any] = None, targets: Optional[List[str]] = None) -> None:
+    def extract(self, path: Optional[Any] = None, targets: Optional[List[str]] = None,
```

Comment:
keep `extract(self, path)->None` method signature as is and introduce read(self, path) -> Dict[str, IO[Any]]` Introduce internal fucntion `_extract(self, path, targets, return_dict) -> Optional[Dict]]` that is your proposal.

That keeps method signature and introduce `read()`